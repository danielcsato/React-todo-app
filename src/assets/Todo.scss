$bg-color: white;

.todo {
  background-color: $bg-color;
  width: 520px;
  padding: 15px;
  border-radius: 10px;
  margin: 10px 0px;
}

.todoTop,
.icons {
  display: flex;
  justify-content: space-between;
}

.icons {
  width: 70px;
}
.icon {
  cursor: pointer;
  font-size: 19px;
}

.todo > p {
  margin-top: 5px;
}
