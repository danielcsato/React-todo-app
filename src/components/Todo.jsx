import React, { useState } from 'react';
import '../assets/Todo.scss';
import PropTypes from 'prop-types';

import { FaTrashAlt } from 'react-icons/fa';
import { FiPlus, FiMinus } from 'react-icons/fi';
import { BiShow, BiHide } from 'react-icons/bi';
import { GrCheckbox, GrCheckboxSelected } from 'react-icons/gr';

import TodoForm from './TodoForm';
import SubTodo from './SubTodo';
import Modal from './Modal';

const Todo = ({ name, id, handleComplete, deleteTodo, isDone, subTasks }) => {
  const [isOpen, setIsOpen] = useState(false);
  const [addNew, setAddNew] = useState(false);
  const [showSubtasks, setShowSubtasks] = useState(true);
  return (
    <div>
      {isOpen && (
        <Modal
          setIsOpen={setIsOpen}
          deleteTodo={() => deleteTodo(id, setIsOpen)}
        />
      )}
      <div
        className={isDone ? 'todoMainDone' : 'todoMain'}
        data-test-id="parentTodoContainer"
      >
        <div
          className={isDone ? 'checkboxDone' : 'checkbox'}
          onClick={() => handleComplete(id)}
        >
          {isDone ? (
            <GrCheckboxSelected className="Undo" />
          ) : (
            <GrCheckbox className="done" />
          )}
        </div>
        <div className="title" id="title" onClick={() => handleComplete(id)}>
          <p>
            {name} {subTasks.length !== 0 && `(${subTasks.length})`}
          </p>
        </div>
        <div className="icons">
          <div className="showSubs">
            {addNew ? (
              <FiMinus
                onClick={() => setAddNew(!addNew)}
                data-test-id="addNewSubTodoMinusIcon"
              />
            ) : (
              <FiPlus
                onClick={() => setAddNew(!addNew)}
                data-test-id="addNewSubTodoIcon"
              />
            )}
          </div>
          {showSubtasks ? (
            <BiHide onClick={() => setShowSubtasks(!showSubtasks)} />
          ) : (
            <BiShow onClick={() => setShowSubtasks(!showSubtasks)} />
          )}

          <FaTrashAlt
            className="trash"
            onClick={() => setIsOpen(true)}
            data-test-id="parentTrashIcon"
          />
        </div>
      </div>
      {addNew && <TodoForm parent={id} subTasks={subTasks} />}
      {showSubtasks && (
        <div>
          {subTasks.length > 0 && (
            <div>
              <ul data-test-id="subtodoUl">
                {subTasks.map((sub) => {
                  return (
                    <li key={sub.id} data-test-id="subtodoLi">
                      <SubTodo
                        name={sub.title}
                        id={sub.id}
                        isDone={sub.isDone}
                        parentId={sub.parentId}
                      />
                    </li>
                  );
                })}
              </ul>
            </div>
          )}
        </div>
      )}
    </div>
  );
};

Todo.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  handleComplete: PropTypes.func,
  deleteTodo: PropTypes.func,
  subTasks: PropTypes.array,
};

export default Todo;
