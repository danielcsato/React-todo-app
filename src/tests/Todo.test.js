// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import React from 'react';
import { shallow } from 'enzyme';

import Todo from '../components/Todo';

describe('Todo.jsx', () => {
  it('Todo is rendered', () => {
    const wrapper = shallow(<Todo />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Todo name is rendered', () => {
    const todo = ['Task'];
    const wrapper = shallow(<Todo name={todo} />);
    expect(wrapper).toMatchSnapshot();
  });
});
