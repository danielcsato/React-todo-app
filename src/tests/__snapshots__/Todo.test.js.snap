// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo.jsx Todo is rendered 1`] = `ShallowWrapper {}`;

exports[`Todo.jsx Todo name is rendered 1`] = `ShallowWrapper {}`;
